src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -52,22 +58,29 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             var signInResult = await _loginService.SignInAsync(viewModel.EmailAddress, viewModel.Password, viewModel.ReturnUrl);
 
             LoginViewModel NewLoginViewModel() =>
-                new LoginViewModel { ReturnUrl = viewModel.ReturnUrl, EmailAddress = signInResult.LoginHint };
+                new LoginViewModel { ReturnUrl = viewModel.ReturnUrl, EmailAddress = signInResult.Value?.LoginHint };
 
             if (!ModelState.IsValid)
                 return View(NewLoginViewModel());
 
-            if (!signInResult.IsSuccessful)
+            if (!signInResult.IsSuccess)
             {
-                ModelState.AddModelError(nameof(LoginViewModel.LoginError), SignInErrorMessage);
+                if (signInResult.Errors.Contains(LoginUserErrors.UserNameOrPasswordIncorrect()))
+                {
+                    ModelState.AddModelError(nameof(LoginViewModel.LoginError), SignInErrorMessage);
+                }
+                if (signInResult.Errors.Contains(LoginUserErrors.UserIsDisabled()))
```

Comment:
What would happen if additional errors were added to the `signInResult` variable from the login service? The next developer would have to know to add additional code here to include that error in the UI. Maybe that is acceptable but trying to open discussion to consider whether it would be beneficial to add any other errors to the `ModelState` or provide better mapping by using the 'Field' property of the error message.